sql/telemetry_derived/glam_client_probe_counts_extract_v1/query.sql

```diff
@@ -1,49 +1,37 @@
--- TODO: Remove deduping when dupes are fixed.
-WITH deduped AS (
-    SELECT
-        *,
-        ROW_NUMBER() OVER(
-            PARTITION BY
-                channel, app_version, agg_type, os, app_build_id, process, metric, key, client_agg_type, metric_type
-            ORDER BY
-                total_users DESC
-        ) AS rank
-    FROM
-        `moz-fx-data-shared-prod.telemetry.client_probe_counts`
-    WHERE
-        channel = @channel
-        AND app_version IS NOT NULL
-        AND total_users > 1000
+CREATE TEMP FUNCTION udf_js_flatten(
+    histogram ARRAY<STRUCT<key STRING, value FLOAT64>>
 )
+RETURNS STRING
+LANGUAGE js
+AS
+'''
+    let data = histogram.map(function(r) {
+        var obj = {};
+        obj[r.key] = parseFloat(r.value.toFixed(4));
+        return obj;
+    });
+    return JSON.stringify(data);
+''';
 
 SELECT
-    CASE
-        WHEN channel="nightly" THEN 1
-        WHEN channel="beta" THEN 2
-        WHEN channel="release" THEN 3
-    END AS channel,
     app_version,
-    CASE
-        WHEN agg_type="histogram" THEN 1
-        WHEN agg_type="percentiles" THEN 2
-    END AS agg_type,
     COALESCE(os, "*") AS os,
     COALESCE(app_build_id, "*") AS app_build_id,
-    CASE
-        WHEN process IS NULL THEN 0
-        WHEN process="parent" THEN 1
-        WHEN process="content" THEN 2
-        WHEN process="gpu" THEN 3
-    END AS process,
+    process,
     metric,
     -- BigQuery has some null unicode characters which Postgresql doesn't like, so we remove those here.
     -- Also limit string length to 200 to match column length.
     SUBSTR(REPLACE(key, r"\x00", ""), 0, 200) AS key,
     client_agg_type,
     metric_type,
     total_users,
-    TO_JSON_STRING(aggregates) AS aggregates
+    MAX(IF(agg_type="histogram", udf_js_flatten(aggregates), NULL)) AS histogram,
```

Comment:
I assume the `MAX()` is so we return the non-null value which is the correct value for `histogram`? Maybe just a comment so it's clear.

Reply:
A comment would be good, I agree. I'm using this as if it's a `COALESCE` but b/c of the `GROUP BY` I need an aggregation function to include this column here. I'll make a note of it.